libevmasm/Instruction.cpp

```diff
@@ -326,12 +327,9 @@ void dev::solidity::eachInstruction(
 		if (isValidInstruction(instr))
 			additional = instructionInfo(instr).additional;
 		u256 data;
-		for (size_t i = 0; i < additional; ++i)
-		{
-			data <<= 8;
-			if (++it < _mem.end())
-				data |= *it;
-		}
+		for (size_t i = 0; i < std::min(additional, static_cast<size_t>(std::distance(it, _mem.end())) - 1); ++i, --additional)
```

Comment:
Hm, this looks really complicated. Would it be easier to do `data <<= 8 * additional` first and then "or in" the iterated values in the loop? I don't feel too happy about modifying `additional`.

Reply:
Then the iterated values would have to be casted to ``u256`` and shifted in the loop, so that they are "or'ed in" in the right places... not sure whether that's any better in the end. I also think that the the ``-1`` in the inner ``for`` might cause us to miss the last value of ``_mem``...
So far I think the lazy option to break the outer loop, if we're already at the end of ``_mem`` is actually the most readable one and it will definitely not change the behaviour.